pkgs/webui.nix

```diff
@@ -2,157 +2,267 @@
+      #
       ./generate-files.sh 'dev-env' webController
+      
+      #
+      # Copy/generate js/css/asset files to/in `./webroot`
+      #
       ./dev-install.sh
       rm -f built_version
+
+      #
+      # Run reqquirejs/r.js ('prod-env' argument is unused)
+      #
       ./build-files.sh 'prod-env' webController
+
+      #
+      # Replace "prod_env" / "dev_env" and add time stamps (all in comments)
+      #
       ./prod-dev.sh webroot/html/dashboard.html prod_env dev_env true
       ./prod-dev.sh webroot/html/login.html prod_env dev_env true
       ./prod-dev.sh webroot/html/login-error.html prod_env dev_env true
-      # end build
 
-      # patch webserver so it doesn't try to write file in the package dir
+      #
+      # copy files which `doPreStartServer()` in webServerStart.js  would otherwise copy (see patchPhase)
+      #
       cp webroot/img/opencontrail-favicon.ico webroot/img/sdn-favicon.ico
       cp webroot/img/opencontrail-logo.png webroot/img/sdn-logo.png
-      substituteInPlace webServerStart.js --replace \
-        "doPreStartServer(false);" \
-        "" \
-      # files generated by the webserver
+
+      #
+      # symlink files to writeable paths outside of the store
+      #
       ln -s /tmp/contrail-web-core-regions.js webroot/common/api/regions.js
       ln -s /tmp/contrail-web-core-menu_wc.xml webroot/menu_wc.xml
-
-      # allow user to provide configuration in /tmp/contrail-web-core-config.js
-      rm -f config/config.global.js
+      rm -rf config/config.global.js
       ln -s /tmp/contrail-web-core-config.js config/config.global.js
 
+      #
+      # generate keys required by jobserver
+      #
+      chmod +x ./generate-keys.sh
+      ./generate-keys.sh
+
       cd ..
     '';
 
     installPhase = ''
       mkdir -p $out
       cp -r * $out
     '';
-    };
-
-    webController = stdenv.mkDerivation {
-      name = "contrail-web-controller";
-      version = "3.2";
-
-      src = webBuild;
-      phases = [ "unpackPhase" "installPhase" ];
-
-      installPhase = ''
-        mkdir $out; cp -r contrail-web-controller/* $out
-      '';
-    };
-    
-    webCore = stdenv.mkDerivation {
-      name = "contrail-web-core";
-      version = "3.2";
+  };
 
-      src = webBuild;
-      phases = [ "unpackPhase" "installPhase" ];
+  webController = stdenv.mkDerivation {
+    name = "contrail-web-controller";
+    version = "5.0";
+    src = webui-build;
+    installPhase = ''
+      mkdir $out
+      cp -r contrail-web-controller/* $out
+    '';
+  };
 
-      installPhase = ''
-        mkdir $out; cp -r contrail-web-core/* $out
-      '';
-    };
+  webCore = stdenv.mkDerivation {
+    name = "contrail-web-core";
+    version = "5.0";
+    src = webui-build;
+    installPhase = ''
+      mkdir $out
+      cp -r contrail-web-core/* $out
+    '';
+  };
 
+in
+  {
+    webui-core = webCore;
+    webui-controller = webController;
```

Comment:
not sure why you use kebab case, we use camel case for all other contrail packages

Reply:
Oh, you are right.. not sure what i was thinking. I'll change it ;-}